zeeguu_api/api/search.py

```diff
@@ -0,0 +1,172 @@
+import flask
+import zeeguu
+from zeeguu.model.search import Search
+from zeeguu.model.search_filter import SearchFilter
+from zeeguu.model.search_subscription import SearchSubscription
+from zeeguu.content_recommender.mixed_recommender import article_search_for_user
+
+from .utils.route_wrappers import cross_domain, with_session
+from .utils.json_result import json_result
+from . import api
+
+session = zeeguu.db.session
+
+SEARCH = "search"
+SUBSCRIBE_SEARCH = "subscribe_search"
+UNSUBSCRIBE_SEARCH = "unsubscribe_search"
+SUBSCRIBED_SEARCHES = "subscribed_searches"
+FILTER_SEARCH = "filter_search"
+UNFILTER_SEARCH = "unfilter_search"
+FILTERED_SEARCHES = "filtered_searches"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{SUBSCRIBE_SEARCH}/<search_terms>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def subscribe_to_search(search_terms):
+    """
+    :param: search_terms -- the search terms to be subscribed to.
+    Subscribe to a search with given search terms.
+
+    """
+    search = Search.find_or_create(session, search_terms)
+    SearchSubscription.find_or_create(session, flask.g.user, search)
+
+    return json_result(search.as_dictionary())
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{UNSUBSCRIBE_SEARCH}/<search_id>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def unsubscribe_from_search(search_id):
+    """
+    A user can unsubscribe from the search with a given ID
+    :return: OK / ERROR
+    """
+    try:
+        to_delete = SearchSubscription.with_search_id(search_id, flask.g.user)
+        session.delete(to_delete)
+        to_delete2 = Search.find_by_id(search_id)
+        session.delete(to_delete2)
+        session.commit()
+
+    except Exception as e:
+        return "OOPS. SEARCH AIN'T THERE IT SEEMS (" + str(e) + ")"
```

Comment:
put more info for the exception in the log: 

   zeeguu.log(e...) 

should think well about exception handling in the API
open questions:
- should we return a 500 code instead of 200 
-